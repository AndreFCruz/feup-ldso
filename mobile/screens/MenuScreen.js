import React from 'react';
import PropTypes from 'prop-types';
import {
    View,
    StyleSheet,
    FlatList,
    Text
} from 'react-native';
import { connect } from 'react-redux';

import { clearFaculty } from '../reducers/modules/facultyReducer';
import { FacultyButton } from '../components/FacultyButton';
import { logos } from '../constants/Logos';

class MenuScreen extends React.Component {
  static navigationOptions = {
      header: null,
  };

  componentDidMount() {
      this.props.navigation.addListener('willFocus', () => this.props.clearFaculty());
  }

  render() {
      return (
          <View style={styles.container}>
              <Text style={styles.header}>ImpactUP</Text>
        
              <FlatList
                  noColumns={3}
                  data={Object.keys(logos).map(elem => {
                      let obj = {};
                      obj.key = elem;
                      return obj;
                  })}
                  renderItem={({ item }) => (
                      <FacultyButton name={item.key} onPress={() => this.props.navigation.navigate('Faculty', {
                          faculty: item.key.toLowerCase()
                      })} />
                  )} />
          </View>
      );
  }
  
}

MenuScreen.propTypes = {
    navigation: PropTypes.object,
    clearFaculty: PropTypes.func
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },

    header: {
        fontSize: 55,
        margin: 20
    }
});

const mapDispatchToProps = {
    clearFaculty
};

export default connect(null, mapDispatchToProps)(MenuScreen);
