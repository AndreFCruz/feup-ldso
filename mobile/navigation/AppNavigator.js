import React from 'react';
import { createStackNavigator } from 'react-navigation';

import MenuScreen from '../screens/MenuScreen';
import FacultyScreen from '../screens/FacultyScreen';
import VideosScreen from '../screens/VideosScreen';
import SocialProjectsScreen from '../screens/SocialProjectsScreen';
import FutureProspectsScreen from '../screens/FutureProspectsScreen';
import LocalizationScreen from '../screens/LocalizationScreen';
import LanguageBtn from '../components/LanguageBtn';

export default createStackNavigator({
  Menu: {
    screen: MenuScreen
  },
  Faculty: {
    screen: FacultyScreen,
    navigationOptions: _ => languageBtn
  },
  Videos: {
    screen: VideosScreen,
    navigationOptions: _ => languageBtn
  },
  SocialProjects: {
    screen: SocialProjectsScreen,
    navigationOptions: _ => languageBtn
  },
  FutureProspects: {
    screen: FutureProspectsScreen,
    navigationOptions: _ => languageBtn
  },
  Localization: {
    screen: LocalizationScreen,
    navigationOptions: _ => languageBtn
  }
}, {
  initialRouteName: 'Menu'
});

var languageBtn = ({
  headerRight: (
    <LanguageBtn/>
  )
});
