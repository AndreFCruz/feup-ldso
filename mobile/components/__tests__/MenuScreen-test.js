import 'react-native';
import React from 'react';
import MenuScreen from '../../screens/MenuScreen';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import AppNavigator from '../../navigation/AppNavigator';
import { FlatList } from 'react-native';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = { };

function setup() {
    const navigation = shallow(<AppNavigator />);

    const props = {
        navigation: navigation
    };

    const menuWrapper = shallow(<MenuScreen {...props} />,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: true
        });

    return {
        props,
        menuWrapper
    };
}

it('renders logo list', () => {
    const { menuWrapper } = setup();

    expect(menuWrapper.dive().find(FlatList).exists()).toEqual(true);
});

it('renders as expected', () => {
    const { menuWrapper } = setup();

    expect(menuWrapper.dive()).toMatchSnapshot();
});
