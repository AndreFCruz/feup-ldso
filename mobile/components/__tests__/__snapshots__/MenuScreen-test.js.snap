// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuScreen
    clearFaculty={[Function]}
    navigation={
      ShallowWrapper {
        Symbol(enzyme.__root__): [Circular],
        Symbol(enzyme.__unrendered__): <NavigationContainer />,
        Symbol(enzyme.__renderer__): Object {
          "batchedUpdates": [Function],
          "getNode": [Function],
          "render": [Function],
          "simulateError": [Function],
          "simulateEvent": [Function],
          "unmount": [Function],
        },
        Symbol(enzyme.__node__): Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navigation": Object {
              "actions": Object {
                "dismiss": [Function],
                "pop": [Function],
                "popToTop": [Function],
                "push": [Function],
                "replace": [Function],
                "reset": [Function],
              },
              "addListener": [Function],
              "dangerouslyGetParent": [Function],
              "dismiss": [Function],
              "dispatch": [Function],
              "getChildNavigation": [Function],
              "getScreenProps": [Function],
              "goBack": [Function],
              "isFocused": [Function],
              "navigate": [Function],
              "pop": [Function],
              "popToTop": [Function],
              "push": [Function],
              "replace": [Function],
              "reset": [Function],
              "router": Object {
                "childRouters": Object {
                  "Faculty": null,
                  "FutureProspects": null,
                  "Localization": null,
                  "Menu": null,
                  "SocialProjects": null,
                  "Videos": null,
                },
                "getActionCreators": [Function],
                "getActionForPathAndParams": [Function],
                "getComponentForRouteName": [Function],
                "getComponentForState": [Function],
                "getPathAndParamsForState": [Function],
                "getScreenOptions": [Function],
                "getStateForAction": [Function],
              },
              "setParams": [Function],
              "state": Object {
                "index": 0,
                "isTransitioning": false,
                "key": "StackRouterRoot",
                "routes": Array [
                  Object {
                    "key": "id-1540648962175-1",
                    "routeName": "Menu",
                  },
                ],
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Symbol(enzyme.__nodes__): Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "navigation": Object {
                "actions": Object {
                  "dismiss": [Function],
                  "pop": [Function],
                  "popToTop": [Function],
                  "push": [Function],
                  "replace": [Function],
                  "reset": [Function],
                },
                "addListener": [Function],
                "dangerouslyGetParent": [Function],
                "dismiss": [Function],
                "dispatch": [Function],
                "getChildNavigation": [Function],
                "getScreenProps": [Function],
                "goBack": [Function],
                "isFocused": [Function],
                "navigate": [Function],
                "pop": [Function],
                "popToTop": [Function],
                "push": [Function],
                "replace": [Function],
                "reset": [Function],
                "router": Object {
                  "childRouters": Object {
                    "Faculty": null,
                    "FutureProspects": null,
                    "Localization": null,
                    "Menu": null,
                    "SocialProjects": null,
                    "Videos": null,
                  },
                  "getActionCreators": [Function],
                  "getActionForPathAndParams": [Function],
                  "getComponentForRouteName": [Function],
                  "getComponentForState": [Function],
                  "getPathAndParamsForState": [Function],
                  "getScreenOptions": [Function],
                  "getStateForAction": [Function],
                },
                "setParams": [Function],
                "state": Object {
                  "index": 0,
                  "isTransitioning": false,
                  "key": "StackRouterRoot",
                  "routes": Array [
                    Object {
                      "key": "id-1540648962175-1",
                      "routeName": "Menu",
                    },
                  ],
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        Symbol(enzyme.__options__): Object {
          "adapter": ReactSixteenAdapter {
            "options": Object {
              "enableComponentDidUpdateOnSetState": true,
              "lifecycles": Object {
                "componentDidUpdate": Object {
                  "onSetState": true,
                },
                "getDerivedStateFromProps": true,
                "getSnapshotBeforeUpdate": true,
                "setState": Object {
                  "skipsComponentDidUpdateOnNullish": true,
                },
              },
            },
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 55,
              "margin": 20,
            }
          }
        >
          ImpactUP
        </Text>,
        <FlatList
          data={
            Array [
              Object {
                "key": "FADEUP",
              },
              Object {
                "key": "FAUP",
              },
              Object {
                "key": "FBAUP",
              },
              Object {
                "key": "FCNAUP",
              },
              Object {
                "key": "FCUP",
              },
              Object {
                "key": "FDUP",
              },
              Object {
                "key": "FEP",
              },
              Object {
                "key": "FEUP",
              },
              Object {
                "key": "FFUP",
              },
              Object {
                "key": "FLUP",
              },
              Object {
                "key": "FMDUP",
              },
              Object {
                "key": "FMUP",
              },
              Object {
                "key": "FPCEUP",
              },
              Object {
                "key": "ICBAS",
              },
            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          noColumns={3}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "ImpactUP",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 55,
            "margin": 20,
          },
        },
        "ref": null,
        "rendered": "ImpactUP",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "key": "FADEUP",
            },
            Object {
              "key": "FAUP",
            },
            Object {
              "key": "FBAUP",
            },
            Object {
              "key": "FCNAUP",
            },
            Object {
              "key": "FCUP",
            },
            Object {
              "key": "FDUP",
            },
            Object {
              "key": "FEP",
            },
            Object {
              "key": "FEUP",
            },
            Object {
              "key": "FFUP",
            },
            Object {
              "key": "FLUP",
            },
            Object {
              "key": "FMDUP",
            },
            Object {
              "key": "FMUP",
            },
            Object {
              "key": "FPCEUP",
            },
            Object {
              "key": "ICBAS",
            },
          ],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "noColumns": 3,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 55,
                "margin": 20,
              }
            }
          >
            ImpactUP
          </Text>,
          <FlatList
            data={
              Array [
                Object {
                  "key": "FADEUP",
                },
                Object {
                  "key": "FAUP",
                },
                Object {
                  "key": "FBAUP",
                },
                Object {
                  "key": "FCNAUP",
                },
                Object {
                  "key": "FCUP",
                },
                Object {
                  "key": "FDUP",
                },
                Object {
                  "key": "FEP",
                },
                Object {
                  "key": "FEUP",
                },
                Object {
                  "key": "FFUP",
                },
                Object {
                  "key": "FLUP",
                },
                Object {
                  "key": "FMDUP",
                },
                Object {
                  "key": "FMUP",
                },
                Object {
                  "key": "FPCEUP",
                },
                Object {
                  "key": "ICBAS",
                },
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            noColumns={3}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "ImpactUP",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 55,
              "margin": 20,
            },
          },
          "ref": null,
          "rendered": "ImpactUP",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "key": "FADEUP",
              },
              Object {
                "key": "FAUP",
              },
              Object {
                "key": "FBAUP",
              },
              Object {
                "key": "FCNAUP",
              },
              Object {
                "key": "FCUP",
              },
              Object {
                "key": "FDUP",
              },
              Object {
                "key": "FEP",
              },
              Object {
                "key": "FEUP",
              },
              Object {
                "key": "FFUP",
              },
              Object {
                "key": "FLUP",
              },
              Object {
                "key": "FMDUP",
              },
              Object {
                "key": "FMUP",
              },
              Object {
                "key": "FPCEUP",
              },
              Object {
                "key": "ICBAS",
              },
            ],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "noColumns": 3,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
